### PR TITLE
refactor(SinginStoreWallet): code

### PR DESCRIPTION
**Pull request by Crazy-MIXER**
- Transferring the screen logic to components
- Move text to i18n
- Replace navigation hooks
- Change the import to absolute

### CODE REVIEW

File: src/components/singin-store-wallet/singin-store-wallet.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import {ETH_HD_SHORT_PATH, MAIN_ACCOUNT_NAME} from '@app/variables';
+
+export const SigninStoreWallet = () => {
+  const navigation = useTypedNavigation();
+  const {nextScreen, mnemonic, privateKey} =
+    useTypedRoute<'restoreStore'>().params;
+  const wallets = useWallets();
+
+  useEffect(() => {
+    showModal(getText(I18N.singinStoreWalletLoading), {
```
Comment (gHashTag) [inline src/components/singin-store-wallet/singin-store-wallet.tsx:20]: The first argument is the name of the modal window that needs to be shown, it should not be a translation

File: src/components/singin-store-wallet/singin-store-wallet.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    setTimeout(async () => {
+      try {
+        if (mnemonic) {
+          let canNext = true;
+          let index = 0;
+          while (canNext) {
+            const name =
+              wallets.getSize() === 0
+                ? MAIN_ACCOUNT_NAME
+                : `Account #${wallets.getSize() + 1}`;
```
Comment (gHashTag) [inline src/components/singin-store-wallet/singin-store-wallet.tsx:38]: Pls add translation

File: src/components/singin-store-wallet/singin-store-wallet.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+              }
+            }
+
+            index += 1;
+          }
+        } else if (privateKey) {
+          const name =
+            wallets.getSize() === 0
+              ? MAIN_ACCOUNT_NAME
+              : `Account #${wallets.getSize() + 1}`;
```
Comment (gHashTag) [inline src/components/singin-store-wallet/singin-store-wallet.tsx:83]: Pls add translation

**Review by gHashTag — CHANGES_REQUESTED** [review]
Fix comments

File: src/components/singin-store-wallet/singin-store-wallet.tsx (hunk 3 above)
Comment (Crazy-MIXER) [reply src/components/singin-store-wallet/singin-store-wallet.tsx:83]: fix all

**Comment by gHashTag** [comment]
I asked for a translation for line `Account #${wallets.getSize() + 1}`, not what you did

**Comment by Crazy-MIXER** [comment]
oh no, I fix fixes

**Review by gHashTag — APPROVED** [review]

File: src/components/singin-store-wallet/singin-store-wallet.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+          default:
+            if (error instanceof Error) {
+              showModal('error-create-account');
+              captureException(error, 'restoreStore');
+              goBack();
+            }
+        }
+      }
+    }, 350);
+  }, [navigation, nextScreen, mnemonic, privateKey, wallets]);
```
Comment (vivalaakam) [inline src/components/singin-store-wallet/singin-store-wallet.tsx:110]: these logic should be on screen, not in component

Comment (Crazy-MIXER) [reply src/components/singin-store-wallet/singin-store-wallet.tsx]: done

**Comment by Crazy-MIXER** [comment]
wait correction in src/i18n.ts

**Pull request merged**